syntax = "proto3";

package protos;

service HybridBookRetriever {
    rpc Retrieve (RetrieveRequest) returns (RetrieveResponse);
}

message RetrieveRequest {
    string query = 1;
    int32 dense_top_k = 2;
    int32 sparse_top_k = 3;
    int32 top_k = 4;
    int32 top_n = 5;
}

message RetrieveResponse {
    repeated string book_ids = 1;
    repeated float scores = 2;
}